azurerm/resource_arm_private_link_service.go

```diff
@@ -0,0 +1,649 @@
+	if len(input) == 0 {
+		return nil
+	}
+
+	results := make([]network.PrivateLinkServiceIPConfiguration, 0)
+
+	for _, item := range input {
+		v := item.(map[string]interface{})
+		privateIpAddress := v["private_ip_address"].(string)
+		privateIPAllocationMethod := v["private_ip_allocation_method"].(string)
+		subnetId := v["subnet_id"].(string)
+		privateIpAddressVersion := v["private_ip_address_version"].(string)
+		name := v["name"].(string)
+		primary := v["primary"].(bool)
+
+		result := network.PrivateLinkServiceIPConfiguration{
+			Name: utils.String(name),
+			PrivateLinkServiceIPConfigurationProperties: &network.PrivateLinkServiceIPConfigurationProperties{
+				PrivateIPAddress:          utils.String(privateIpAddress),
+				PrivateIPAddressVersion:   network.IPVersion(privateIpAddressVersion),
+				PrivateIPAllocationMethod: network.IPAllocationMethod(privateIPAllocationMethod),
+				Subnet: &network.Subnet{
+					ID: utils.String(subnetId),
+				},
+				Primary: utils.Bool(primary),
+			},
+		}
+
+		results = append(results, result)
+	}
+	return &results
+}
+
+func expandArmPrivateLinkServiceFrontendIPConfiguration(input []interface{}) *[]network.FrontendIPConfiguration {
+	if len(input) == 0 {
+		return nil
+	}
+
+	results := make([]network.FrontendIPConfiguration, 0)
+
+	for _, item := range input {
+		result := network.FrontendIPConfiguration{
+			ID: utils.String(item.(string)),
+		}
+
+		results = append(results, result)
+	}
+
+	return &results
+}
+
+func expandArmPrivateLinkServicePrivateEndpointConnection(input []interface{}) *[]network.PrivateEndpointConnection {
+	results := make([]network.PrivateEndpointConnection, 0)
+	for _, item := range input {
+		v := item.(map[string]interface{})
+		id := v["id"].(string)
+		privateEndpoint := v["private_endpoint"].([]interface{})
+		privateLinkServiceConnectionState := v["private_link_service_connection_state"].([]interface{})
+		name := v["name"].(string)
+
+		result := network.PrivateEndpointConnection{
+			ID:   utils.String(id),
+			Name: utils.String(name),
+			PrivateEndpointConnectionProperties: &network.PrivateEndpointConnectionProperties{
+				PrivateEndpoint:                   expandArmPrivateLinkServicePrivateEndpoint(privateEndpoint),
+				PrivateLinkServiceConnectionState: expandArmPrivateLinkServicePrivateLinkServiceConnectionState(privateLinkServiceConnectionState),
+			},
+		}
+
+		results = append(results, result)
+	}
+	return &results
+}
+
+func expandArmPrivateLinkServicePrivateLinkServicePropertiesVisibility(input []interface{}) *network.PrivateLinkServicePropertiesVisibility {
+	if len(input) == 0 {
+		return nil
+	}
+
+	subscriptions := make([]string, 0)
+
+	for _, v := range input {
+		subscriptions = append(subscriptions, v.(string))
+	}
+
+	result := network.PrivateLinkServicePropertiesVisibility{
+		Subscriptions: &subscriptions,
```

Comment:
As above we can use utils.expandstringslice and set this inline

Reply:
Fixed, but I kept the function as explained above.